### PR TITLE
Windows: Initial compilation enablement

### PR DESCRIPTION
**Pull request by jhowardmsft**
Signed-off-by: John Howard <jhoward@microsoft.com>
@swernli

This PR is part of the proposal described in issue docker/docker 10662 to port the docker daemon to Windows. While it touches a large number of files, this is the minimum set of changes to allow libcontainer to build on Windows. It adds build tags as appropriate, factors out cgroups (not applicable on Windows), adds some TODOs for future Windows refactoring, factors out namespaces (not applicable on Windows either) and adds a dummy console on Windows.

### CODE REVIEW

File: cgroups/cgroups_unsupported.go [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// +build windows
```
Comment (crosbymichael) [inline cgroups/cgroups_unsupported.go]: Should things like this be `!linux` ?

Comment (jhowardmsft) [inline cgroups/cgroups_unsupported.go]: Updated (also fs_unsupported.go and sysconfig.go)

**Comment by jhowardmsft** [comment]
Note before libcontainer in docker/docker is "re-vendored" with this change, the following PR must be taken into the docker/docker codebase: https://github.com/docker/docker/pull/13269

**Comment by jhowardmsft** [comment]
@crosbymichael @tianon - note above comment re dependency. Thanks.

File: configs/namespaces_linux.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		file = "ipc"
+	case NEWUSER:
+		file = "user"
+	case NEWUTS:
+		file = "uts"
+	}
+	return file
+}
+
+func (n *Namespaces) Remove(t NamespaceType) bool {
```
Comment (LK4D4) [inline configs/namespaces_linux.go]: I'm not sure, but this methods looks platform independent.

Comment (jhowardmsft) [inline configs/namespaces_linux.go]: You're absolutely right, it is. There are many more changes expected throughout libcontainer for Windows daemon support. The set of changes in this PR is the minimum set I can get away with to at least get the Windows daemon to a compiling state (assuming all my current docker/docker PRs are accepted and libcontainer is re-vendored). The Windows daemon will not actually do anything at this point though. Getting a consistently compiling daemon on Windows (and a build check for it not being broken by Linux changes) is a milestone towards getting a functional daemon as per the //build demos.

Comment (LK4D4) [inline configs/namespaces_linux.go]: Got it, thanks.

**Comment by LK4D4** [comment]
LGTM

**Comment by crosbymichael** [comment]
LGTM

**Comment by crosbymichael** [comment]
thanks @jhowardmsft

**Pull request merged**